### ISSUE TITLE
Same loss different/incorrect results

### ISSUE THREAD

**Issue by pbamotra** [tensorflow/tensorflow#20040]
TF version: 1.8
Installation: tensorflow/tensorflow:latest-gpu
OS details: 
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.4 LTS
Release:	16.04
Codename:	xenial

AWS instance: p2.xlarge
GPU details:
```
[name: "/device:CPU:0"
device_type: "CPU"
memory_limit: 268435456
locality {
}
incarnation: 16086859869116902206
, name: "/device:GPU:0"
device_type: "GPU"
memory_limit: 11285974221
locality {
  bus_id: 1
  links {
  }
}
incarnation: 13890740079777279899
physical_device_desc: "device: 0, name: Tesla K80, pci bus id: 0000:00:1e.0, compute capability: 3.7"
]
```

Custom code:
This works
```python
model.compile(loss='sparse_categorical_crossentropy',
              optimizer=keras.optimizers.Adam(lr=0.01, clipnorm=1., amsgrad=True),
              metrics=['acc'])
```
This works too
```python
model.compile(loss=keras.losses.sparse_categorical_crossentropy,
              optimizer=keras.optimizers.Adam(lr=0.01, clipnorm=1., amsgrad=True),
              metrics=['acc'])
```
This doesn't work. The loss value looks fine but the accuracy stays at 0 or very low value.
```python
def sparse_cate_loss(y_true, y_pred):
    return keras.losses.sparse_categorical_crossentropy(y_true, y_pred)

model.compile(loss=sparse_cate_loss,
              optimizer=keras.optimizers.Adam(lr=0.01, clipnorm=1., amsgrad=True),
              metrics=['acc'])
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 44 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
We are closing this issue for now due to lack of activity. Please comment if this is still an issue for you. Thanks!
